pkg/api/types.go

```diff
@@ -1050,15 +1048,18 @@ func (p *Properties) GetVirtualNetworkName() string {
 // GetSubnetName returns the subnet name of the cluster based on its current configuration.
 func (p *Properties) GetSubnetName() string {
```

Comment:
@jackfrancis I fixed the function to look more like the armvariables logic so it's easier to compare and we were indeed missing the condition `if !p.IsHostedMasterProfile() && !p.MasterProfile.IsCustomVNET() && p.MasterProfile.IsVirtualMachineScaleSets()` then `subnetName = "subnetmaster"` 

Hopefully the tests at https://github.com/Azure/aks-engine/pull/2659/files#diff-de120f39e8d8448b9aa430e66c997a97R4820 help bring more confidence that it actually does the same thing...

That being the naming logic doesn't really make sense and I don't see why vmss masters need a different subnet name but I'm guessing that's just the natural evolution of the two templates being in separate files previously, and changing the behavior would mean risking breaking upgrade/scale so I'm going to stay away from that for now.

Reply:
I recommend looking at the file directly https://github.com/Azure/aks-engine/blob/01e8d7720d6483b95467294e2561cdebef260dd0/pkg/api/types.go#L1049, the diff is pretty hard to read...